File: Easy/Recamans sequence.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution
{
public:
    vector<int> recamanSequence(int n)
    {
        // code here
        set<int> mappa;
        vector<int> ans;
        
        ans.push_back(0);
        mappa.insert(0);
        
        for (int i=1; i<n; i++)
        {
            int last = ans[ans.size()-1];
            int el = last - i;
            
            if (el > 0 && mappa.find(el) == mappa.end())
            {
                mappa.insert(el);
                ans.push_back(el);
            }
            else
            {
                mappa.insert(last + i);
                ans.push_back(last + i);
            }
            
        }
        
        return ans;
        
    }
};

//{ Driver Code Starts.

int main(){
    int t;
    cin>>t;
    while(t--){
        int n;
        cin>>n;
        
        Solution ob;
        vector<int> ans = ob.recamanSequence(n);
        for(int i = 0;i < n;i++)
            cout<<ans[i]<<" ";
        cout<<"\n";
    }
    return 0;
}
// } Driver Code Ends

File: Easy/Recamans sequence.py
#User function Template for python3

class Solution:
    def recamanSequence(self, n):
        # code here
        used = set()
        used.add(0)
        
        ans = [0]
        
        last = 0
        
        for i in range(1, n + 1):
            lastmin = last - i
            
            #print(f"i: {i}  ---   last: {last}  ---  lastmin: {lastmin}  ---  lastplus: {last + i}")
            
            if lastmin > 0 and lastmin not in used:
                used.add(lastmin)
                ans.append(lastmin)
                last = lastmin
            else:
                lastmax = last + i
                used.add(lastmax)
                ans.append(lastmax)
                last = lastmax
                
            #print(ans, "\n")
        
        return ans


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())
        
        ob = Solution()
        ans = ob.recamanSequence(n)
        for i in range(n):
            print(ans[i],end=" ")
        print()
# } Driver Code Ends